Lib/test/test_posix.py

```diff
@@ -1476,6 +1478,106 @@ def test_empty_file_actions(self):
         )
         self.assertEqual(os.waitpid(pid, 0), (pid, 0))
 
+    def test_resetids(self):
+        pid = posix.posix_spawn(
+            sys.executable,
+            [sys.executable, '-c', 'pass'],
+            os.environ,
+            resetids=True
+        )
+
+        self.assertEqual(os.waitpid(pid, 0), (pid, 0))
+
+
+    def test_resetids_wrong_type(self):
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, resetids=0)
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, resetids=1.0)
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, resetids="True")
+
+    def test_setpgroup(self):
+        pid = posix.posix_spawn(
+            sys.executable,
+            [sys.executable, '-c', 'pass'],
+            os.environ,
+            setpgroup=os.getpgrp()
+        )
+
+        self.assertEqual(os.waitpid(pid, 0), (pid, 0))
+
+
+    def test_setsigmask_wrong_type(self):
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, setsigmask=["j","k","k"])
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, setsigmask=[3+4j])
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, setsigmask="iuhuhiuh")
+
+    def test_setsigmask(self):
+
+        pid = posix.posix_spawn(
+            sys.executable,
+            [sys.executable, '-c', "pass"],
+            os.environ,
+            setsigmask=[signal.SIGUSR1]
+        )
+
+        self.assertEqual(os.waitpid(pid, 0), (pid, 0))
+
+    def test_setsigdef_wrong_type(self):
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, setsigdef=["j","k","k"])
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, setsigdef=[3+4j])
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, setsigdef="iuhuhiuh")
+
+    def test_setsigdef(self):
```

Comment:
This test has the same problem as the other one. The test that does not work on Travis is:

```python
     def test_setsigdef(self):
         import signal, os
         original_handler = signal.signal(signal.SIGUSR1, signal.SIG_IGN)

         code = textwrap.dedent("""\
              import time
              time.sleep(0.1)""")

         try:
             pid = posix.posix_spawn(
                 sys.executable,
                 [sys.executable, '-c', code],
                 os.environ,
                 setsigdef=[signal.SIGUSR1]
             )

             os.kill(pid, signal.SIGUSR1)
         finally:
             signal.signal(signal.SIGUSR1, original_handler)

         self.assertNotEqual(os.waitpid(pid, 0), (pid, 0))

```